flashbot-attack/src/agent.js

```diff
@@ -1,71 +1,74 @@
-const {
-  Finding,
-  FindingSeverity,
-  FindingType,
-  getEthersProvider,
-} = require('forta-agent');
-const { default: axios } = require('axios');
+const { Finding, FindingSeverity, FindingType, getTransactionReceipt } = require("forta-agent");
+const { default: axios } = require("axios");
 
-const flashbotsUrl = 'https://blocks.flashbots.net/v1/blocks?limit=10';
+const flashbotsUrl = "https://blocks.flashbots.net/v1/blocks?limit=10";
 let lastBlockNumber = 0;
 
-const handleBlock = async () => {
-  let result;
-  try {
-    result = await axios.get(flashbotsUrl);
-  } catch (e) {
-    console.log('Error:', e.code);
-    return [];
-  }
+function provideHandleBlock(getTransactionReceipt) {
+  return async () => {
+    let result;
+    try {
+      result = await axios.get(flashbotsUrl);
+    } catch (e) {
+      console.log("Error:", e.code);
+      return [];
+    }
 
-  const { blocks } = result.data;
+    const { blocks } = result.data;
 
-  // Get findings for every new flashbot block and combine them
-  let findings = await Promise.all(blocks.map(async (block) => {
-    const { transactions, block_number: blockNumber } = block;
-    let currentBlockFindings;
+    // Get findings for every new flashbot block and combine them
+    let findings = await Promise.all(
+      blocks.map(async (block) => {
+        const { transactions, block_number: blockNumber } = block;
+        let currentBlockFindings;
 
-    // Only process blocks that aren't processed
-    if (blockNumber > lastBlockNumber) {
-      // Create finding for every flashbot transaction in the block
-      currentBlockFindings = await Promise.all(transactions.map(async (transaction) => {
-        const {
-          eoa_address: from,
-          to_address: to,
-          transaction_hash: hash,
-        } = transaction;
+        // Only process blocks that aren't processed
+        if (blockNumber > lastBlockNumber) {
+          // Create finding for every flashbot transaction in the block
+          currentBlockFindings = await Promise.all(
+            transactions
+              .filter((tx) => tx.bundle_type !== "mempool")
```

Comment:
```suggestion
              .filter((transaction) => transaction.bundle_type !== "mempool")
```
Renaming suggestion just so that it's the same as the `.map`